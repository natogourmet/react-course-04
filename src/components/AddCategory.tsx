import { useState } from 'react';

type props = {
  onInputSubmit: (category: string) => void;
};

export const AddCategory = ({ onInputSubmit }: props) => {
  const [inputValue, setInputValue] = useState('One Punch');

  const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    if (inputValue.trim().length < 1) return;
    onInputSubmit(inputValue);
    setInputValue('');
  };

  return (
    <form action="" onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Add Category"
        value={inputValue}
        onChange={(e) => setInputValue(e.target.value)}
      />
    </form>
  );
};
