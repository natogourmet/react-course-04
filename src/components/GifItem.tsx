import PropTypes from 'prop-types';

type props = {
  url: string;
  title: string;
}

export const GifItem = ({url, title}: props) => {
  return (
    <div className="card">
      <img src={url} alt="" />
      <p>{title}</p>
    </div>
  )
}

GifItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}