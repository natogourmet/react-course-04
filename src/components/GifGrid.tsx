import { useEffect, useState } from 'react';
import { getGifs } from '../helpers/getGifs';
import { GifItem } from './GifItem';
type props = {
  category: string;
};

export const GifGrid = ({ category }: props) => {
  const [images, setImages] = useState([]);
  
  const getImages = async (category: string) => {
    const images = await getGifs(category);
    setImages(images);
  };

  useEffect(() => {
    getImages(category);
  }, []);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map((img: any) => (
          <GifItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};
