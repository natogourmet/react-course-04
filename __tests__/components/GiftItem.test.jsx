import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('tests on GiftItem component', () => {
  const title = 'A title';
  const url = 'https://localhost/algo.jpg';

  test('should match with snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('should show image with URL and ALT given', () => {
    render(<GifItem title={title} url={url} />);
    screen.debug();

    // expect( screen.getByRole('img').src).toBe( url);
    const { src } = screen.getByRole('img');
    expect(src).toBe(url);
  });

  test('should show the Title', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
