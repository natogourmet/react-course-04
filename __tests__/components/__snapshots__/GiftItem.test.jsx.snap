// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on GiftItem component should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt=""
      src="https://localhost/algo.jpg"
    />
    <p>
      A title
    </p>
  </div>
</div>
`;
